salsa20/src/lib.rs

```diff
@@ -61,6 +61,24 @@
 //! assert_eq!(buffer, ciphertext);
 //! ```
 //!
+//! # Configuration Flags
+//!
+//! You can modify crate using the following configuration flags:
+//!
+//! - `salsa20_force_soft`: force software backend.
+//! - `salsa20_force_sse2`: force SSE2 backend on x86/x86_64 targets.
```

Comment:
Can you leave these changes around a `salsa20_force_sse2` feature out?

You can largely accomplish the same thing via setting the appropriate target features, which will compile out runtime CPU feature detection.

We'd also like to switch to a different categorization of these which would reduce the massive proliferation of `cfg` options across all the crates in our org: https://github.com/RustCrypto/meta/issues/18

Reply:
I agree with that. i Made simplifications at https://github.com/RustCrypto/stream-ciphers/pull/348/commits/840ff5e2b1b6f3997ba3de5d0e4375ebdeea153c

Reply:
Completed at https://github.com/RustCrypto/stream-ciphers/pull/348/commits/ab83323b186d4dac92f81605bb132738fbb1e414. Let me know if i am removing too much.